yt_dlp/YoutubeDL.py

```diff
@@ -1619,17 +1619,26 @@ def extract_info(self, url, download=True, ie_key=None, extra_info=None,
     def _handle_extraction_exceptions(func):
         @functools.wraps(func)
         def wrapper(self, *args, **kwargs):
+            wait_retries = 0
+            max_wait_retries = self.params.get('wait_retries')
             while True:
                 try:
                     return func(self, *args, **kwargs)
                 except (CookieLoadError, DownloadCancelled, LazyList.IndexError, PagedList.IndexError):
                     raise
                 except ReExtractInfo as e:
+                    if wait_retries >= max_wait_retries:
+                        if max_wait_retries > 0:
+                            self.report_error(f'Giving up after {wait_retries} {"retries" if wait_retries > 1 else "retry"} while waiting.')
+                        else:
+                            self.report_error('Video is still unavailable after waiting.')
+                        return
```

Comment:
`ReExtractInfo` does not necessarily imply `--wait-for-video`. This needs to be implemented adjacent to the sleep logic

Reply:
About "adjacent to the sleep logic", I assume you mean in `_wait_for_video()` [here](https://github.com/yt-dlp/yt-dlp/blob/51d7bb8b8cd6342f3578e780a27a198d7767eb37/yt_dlp/YoutubeDL.py#L1690). That seems more obvious, but in that case I'd need to add a new attribute to the class, essentially adding a new bit of global state that has to be reset somewhere up the call stack just so `_wait_for_video()` can count how many times it's been called. (possibly in `extract_info()` [here](https://github.com/yt-dlp/yt-dlp/blob/51d7bb8b8cd6342f3578e780a27a198d7767eb37/yt_dlp/YoutubeDL.py#L1613))

So I can put it closer to the sleep logic, but then it's also going to be farther from the sleep logic, and harder to understand.

Initially, I thought of either subclassing `ReExtractInfo` or adding a field to it that would indicate we're re-extracting because of a timeout. I gave up on that because it wasn't being thrown from anywhere else, so I went with minimum impact on existing code. Though I now notice that `ThrottledDownload` also derives from it. So I think making a new subclass of `ReExtractInfo` would be the right idea.

Reply:
> So I think making a new subclass of `ReExtractInfo` would be the right idea.

That would solve this particular problem.

However, another issue is that this code triggers **after** the wait period, which is not the desired behavior. For example `--wait-retries 1` should do `extract -> wait -> extract -> abort`. Instead, it does `extract -> wait -> extract -> wait -> abort`

I don't believe there's a non-hacky way to resolve this. Instead of a global state though, could we pass something like `{'__attempt': idx}` as `extra_info` to the underlying `__extract_info`, which can then be forwarded to `_wait_for_video`?